config/samples/flows_v1alpha1_flowcollector.yaml

```diff
@@ -4,12 +4,19 @@ metadata:
   name: cluster
 spec:
   namespace: netobserv
+  deploymentType: DIRECT
   agent:
     type: EBPF
```

Comment:
Not 100% related to this task, but I was thinking in the case where a customer would like to override some properties of the eBPF agent, which is the only supported agent for downstream.

They should provide a yaml fragment like:

```yaml
agent:
  ebpf:
    sampling: 1
```

Being `ebpf` the only section they should use, wouldn't it be cleaner just to:
- remove agent.type
- move all the ebpf stuff as properties of the `agent` section
- try to find a "hidden" place for `ipfix` section that would allow overriding the eBPF agent without making our users to "pay" the price of a discriminated union.

Another option could be to move as properties of `agent` these properties that are common to all the agents (sampling, cacheMaxLenght, cacheMaxTimeout...) and keep some implementation details as subsections of `ebpf` (like the image, the image pull policy, etc...)

Reply:
If sampling is common to both agent, can we move this part upper?

Something like :
```
agent:
  sampling: 1
  ebpf:
      ...
  ipfix:
      ...
```

Reply:
I agree that this would make more sense.

Reply:
(Commenting here for the third time because I am reading this too fast...)

My understanding is we are finally keeping both agent downstream but we are going to document clearly that only the ebpf agent is officially supported.

But may be we could move to the agent section the field that are relevent to both agent:
```
agent:
   sampling: 1
   ebpf:
      ...
   ipfix:
      ...
```

Reply:
they are not shared fields, and I don't think we want make them shared, for instance because we want to have different default values (eBPF supports higher sampling rate than IPFIX for instance)